src/main/java/team/creative/creativecore/common/gui/GuiParent.java

```diff
@@ -166,13 +178,49 @@ public GuiChildControl add(GuiControl control) {
         controls.add(child);
         return child;
     }
+
+    public GuiParent addWidget(GuiControl control) {
```

Comment:
Why use addWidget if there is already a method `add` for it. The other methods should follow the same naming scheme. Just call them add. I like the new method variants!

Reply:
I can't call them add because it have the same arguments

the difference between add and addWidget is the return value, but compiler doesn't let you have 2 methods with the same arguments (even if the returned class wasn't the same)

Reply:
![image](https://github.com/CreativeMD/CreativeCore/assets/29090346/1e7e9372-f2c9-4373-9f86-b3bcc6b932d2)

Reply:
renaming `addWidgets(GuiControl... controls)` is possible, but sadly compiler isn't smart enough to understand what signature want to use, internally compiler transforms the arguments into an array instance `new GuiControl { ... }` but when exists other method who matches exactly with the signature the "array" version is skipped and that basically uglifies the code (and that was i am trying to avoid)

Reply:
may can i made a suggestion? rename the original `add` to `addControl` and have the chained variants called as `add`

Reply:
in theory no at all, for simples mods with no overrides on `addControl` to throw an `UnsupportedOperationException` bumping CC and rebuild should be enough

Reply:
Yeah just go for it, I will have to reupload all of my mods then. I might even push out a major version update to make people realize they need to update.

Reply:
with the proper version range should be enough